wordcloud.py

```diff
@@ -1,30 +1,28 @@
+"""Create word cloud tools."""
 import operator
-import random
 import string
 
-FILENAME = "initial_text.txt"
 WORDS_WITH_VALUE = ["word", "cloud", "punctuation"]
 
 
-def main():
-    """Produce words suitable for processing into a word cloud using random model selection."""
-    raw_text = read_file(FILENAME)
+def main(source_text, model_selection="occurrence"):
```

Comment:
I have changed the name of the main() function to word_cloud_logic().  That doesn’t really sound like a verb phrase to me so if you’d like it changed further (eg to create_word_cloud_logic), that’s fine.  I thought this gave a reasonable description though.

Reply:
word_cloud_logic() is more descriptive than main. 👍 (You could use that kind of description in the module docstring)

When figuring out names I try to see how it looks in code that uses it. For example, in wordcloud_test.py it might look something like this:

```word_to_value = word_cloud_logic(text, mode)```

Is it clear from the name what the function does above? If no then it could potentially be improved.

We could do something like this:

```word_to_value = process_text(text, mode)```

But that sounds very generic, which is one reason I think it might be a good idea to have a different function for each model (I discuss this more in another comment). Then you could have names like:

```
word_to_occurence = map_word_to_occurrence(text)
word_to_length = map_word_to_length(text)
word_to_value = map_word_to_alphabetical_order(text)
```

If you agree with this approach I think those are good names that you can use 👍

Reply:
OK thank you - I think I've got them looking better now.